### PR TITLE
feat: support domain at auto mode

### PR DESCRIPTION
**Pull request by Gabriel-403**
Fix: #25, Fix: #66

### CODE REVIEW

**Comment by hsluoyz** [comment]
@nodece @Zxilly @kingiw plz review.

**Review by Sagilio — APPROVED** [review]

File: src/Authorizer.ts [hunk 1]
```diff
@@ -4,6 +4,7 @@ import * as casbin from 'casbin';
 import Permission from './Permission';
 import { StringKV } from './types';
 import * as Cache from './Cache';
+import { stringify } from 'ts-jest/dist/utils/json';
```
Comment (Zxilly) [inline src/Authorizer.ts:7]: why import this?

**Review by Zxilly — CHANGES_REQUESTED** [review]

File: src/Authorizer.ts (hunk 1 above)
Comment (Sagilio) [reply src/Authorizer.ts:7]: Maybe forget to clear

Comment (Gabriel-403) [reply src/Authorizer.ts:7]: Sorry.This is my negligence. Deal with it immediately! @Zxilly @Sagilio

File: src/Authorizer.ts [hunk 2]
```diff
@@ -93,7 +92,14 @@ export class Authorizer {
         this.enforcer = await casbin.newEnforcer(m);
         if ('p' in obj) {
             for (const sArray of obj['p']) {
-                await this.enforcer.addPolicy(sArray[1].trim(), sArray[2].trim(), sArray[3].trim());
+                let arr = sArray as string[];
+                arr = arr.map(v => v.trim())
+                const pType = arr.shift()
+                if (pType == 'p'){
+                    await this.enforcer.addPolicy(...arr);
+                } else if (pType == 'g'){
```
Comment (Zxilly) [inline src/Authorizer.ts:100]: As for now, node-casbin has supoorted multipie Role Manager. It means we will also has `g2,g3...` policies.
It's OK for just supoort one role manager. But plz add a warning when the user got this unhandle situation.

**Review by Zxilly — APPROVED** [review]

File: src/Authorizer.ts (hunk 2 above)
Comment (Gabriel-403) [reply src/Authorizer.ts:100]: Thank you for your suggestion, I will finish it as soon as possible in the next iteration

**Review by nodece — APPROVED** [review]
LGTM

**Comment by Sagilio** [comment]
@nodece @Zxilly @Gabriel-403 
We may also need to add `g`/`g1`/`g2` policies at `CasbinJsGetPermissionForUser` return.
https://github.com/casbin/casbin/blob/aeeb9e6854e26bdbe22145e212b92c930df3036d/frontend.go#L27-L33

**Pull request merged**